### PR TITLE
action: create serverless cluster

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

GitHub action so we can leverage the existing oblt-cli integration to create serverless clusters

## Why is it important?

To use the Github issue automation

### CODE REVIEW

File: .github/workflows/test-oblt-cli-create-serverless.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+name: test-oblt-cli-create-serverless
+
+on:
+  workflow_dispatch: ~
```
Comment (reakaleek) [inline .github/workflows/test-oblt-cli-create-serverless.yml:5]:
```suggestion
  workflow_dispatch: ~
  push:
    branches:
      - main
    paths:
      - '.github/actions/oblt-cli-create-serverless/**'    
```

**Review by reakaleek — APPROVED** [review]

File: .github/actions/oblt-cli-create-serverless/action.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        {
+          [ -n '${{ inputs.cluster-name-prefix }}' ] \
+            && echo -n '--cluster-name-prefix=${{ inputs.cluster-name-prefix }} '
+          [ -n '${{ inputs.cluster-name-suffix }}' ] \
+            && echo -n '--cluster-name-suffix=${{ inputs.cluster-name-suffix }} '
+          [ '${{ inputs.dry-run }}' == 'true' ] \
+            && echo -n '--dry-run '
+        } > .flags
+        echo "COMMAND=$(cat .flags)" >> $GITHUB_ENV
+        rm .flags
```
Comment (kuisathaverat) [inline .github/actions/oblt-cli-create-serverless/action.yml:52]: NIT: this in action script is more clear and about the same length of code.

File: .github/actions/oblt-cli-create-serverless/action.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      run: |
+        {
+          echo -n '{'
+          echo -n '"GitOps":"true",'
+          echo -n '"GitHubRepository":"${{ github.repository }}"'
+          [ -n '${{ github.event.issue.number }}' ] && echo -n ',"GitHubIssue":"${{ github.event.issue.number }}"'
+          echo -n '}'
+        } > .gitops
+        echo "GITOPS=$(cat .gitops)" >> $GITHUB_ENV
+        rm .gitops
```
Comment (kuisathaverat) [inline .github/actions/oblt-cli-create-serverless/action.yml:66]: ditto

File: .github/actions/oblt-cli-create-serverless/action.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      run: |
+        {
+          echo -n '{'
+          echo -n '"GitOps":"true",'
+          echo -n '"GitHubRepository":"${{ github.repository }}",'
+          echo -n '"GitHubCommit":"${{ github.event.comment.id }}",'
+          echo -n '"GitHubPullRequest":"${{ github.event.issue.number }}"'
+          echo -n '}'
+        } > .gitops
+        echo "GITOPS=$(cat .gitops)" >> $GITHUB_ENV
```
Comment (kuisathaverat) [inline .github/actions/oblt-cli-create-serverless/action.yml:80]: ditto

File: .github/actions/oblt-cli-create-serverless/action.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    - name: Configure oblt-cli serverless parameters
+      run: |
+        {
+          echo -n '{'
+          echo -n '"Target":"${{ inputs.target }}",'
+          echo -n '"ProjectType":"${{ inputs.project-type }}"'
+          echo -n '}'
+        } > .serverless
+        echo "SERVERLESS=$(cat .serverless)" >> $GITHUB_ENV
+        rm .serverless
```
Comment (kuisathaverat) [inline .github/actions/oblt-cli-create-serverless/action.yml:93]: ditto

**Review by kuisathaverat — APPROVED** [review]

File: .github/actions/oblt-cli-create-serverless/action.yml (hunk 2 above)
Comment (v1v) [reply .github/actions/oblt-cli-create-serverless/action.yml:52]: Let me revisit this

Comment (v1v) [reply .github/actions/oblt-cli-create-serverless/action.yml:52]: I'll create a follow up

**Pull request merged**